src/main/java/seedu/address/logic/parser/MarkCommandParser.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.MarkCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.StudentId;
+
+/**
+ * Parses input arguments and creates a new MarkCommand object
+ */
+
+public class MarkCommandParser implements Parser<MarkCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the MarkCommand and returns a MarkCommand
+     * object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public MarkCommand parse(String args) throws ParseException {
```

Comment:
I think you can put a @SuppressWarnings("OptionalGetWithoutIsPresent") here since we have checked it with the custom arePrefixesPresent method
```suggestion
    @SuppressWarnings("OptionalGetWithoutIsPresent") // This is checked using our custom function (arePrefixesPresent).
    public MarkCommand parse(String args) throws ParseException {
```